app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/room/viewmodel/VideosViewModel.java

```diff
@@ -0,0 +1,211 @@
+                                mDatabaseHelper
+                                        .getDatabase()
+                                        .videoDao()
+                                        .searchVideos(queryMessage);
+                        if (AppConfiguration.IS_DATABASE_ACCESS_LATENCY_ENABLED
+                                || AppConfiguration.IS_SEARCH_LATENCY_ENABLED) {
+
+                            /**
+                             * Emit the result using specified delay
+                             */
+                            return sendThroughMediatorLiveData(source, 3000L);
+                        } else {
+                            return source;
+                        }
+                    }
+                });
+
+
+        mVideoById = Transformations.switchMap(
+                mVideoId, new Function<Long, LiveData<VideoEntity>>() {
+                    @Override
+                    public LiveData<VideoEntity> apply(final Long videoId) {
+
+                        /**
+                         * Fetching live data from database
+                         */
+                        final LiveData<VideoEntity> source =
+                                mDatabaseHelper
+                                        .getDatabase()
+                                        .videoDao()
+                                        .loadVideoById(videoId);
+
+                        /**
+                         * If latency is enabled, we will return the data using the mediator
+                         * live data
+                         */
+                        if (AppConfiguration.IS_DATABASE_ACCESS_LATENCY_ENABLED) {
+                            return sendThroughMediatorLiveData(source, 2000L);
+                        }
+                        return source;
+                    }
+                });
+
+    }
+
+    public LiveData<List<VideoEntity>> getSearchResult() {
+        return mSearchResults;
+    }
+
+    public LiveData<VideoEntity> getVideoById() {
+        return mVideoById;
+    }
+
+    public LiveData<List<VideoEntity>> getVideosInSameCategory() {
+        return mAllVideosByCategory;
+    }
+
+    public LiveData<List<CategoryEntity>> getAllCategories() {
+        return mAllCategories;
+    }
+
+    public void setQueryMessage(String queryMessage) {
+        mQuery.setValue(queryMessage);
+    }
+
+    public void setVideoId(Long videoIdVal) {
+        mVideoId.setValue(videoIdVal);
+    }
+
+    public void setCategory(String category) {
+        mVideoCategory.setValue(category);
+    }
+
+    /**
+     * Helper function to use mediator live data to emit the live data using specified delay
+     *
+     * @param source source live data
+     * @param ms     for delay
+     * @param <T>    The type of data you want to emit in the live data
+     * @return The mediator live data
+     */
+    private <T> MediatorLiveData<T> sendThroughMediatorLiveData(LiveData<T> source, final Long ms) {
```

Comment:
seems like this is copied between VMs, better move to a utility class.

Reply:
this method is removed for this version